x-pack/legacy/plugins/lens/public/types.ts

```diff
@@ -182,31 +177,59 @@ export interface DatasourceDataPanelProps<T = unknown> {
   filters: Filter[];
 }
 
-// The only way a visualization has to restrict the query building
-export interface DatasourceDimensionPanelProps {
+export interface DatasourceDimensionProps<T> {
   layerId: string;
   columnId: string;
 
-  dragDropContext: DragContextState;
-
   // Visualizations can restrict operations based on their own rules
   filterOperations: (operation: OperationMetadata) => boolean;
 
   // Visualizations can hint at the role this dimension would play, which
   // affects the default ordering of the query
   suggestedPriority?: DimensionPriority;
+
   onRemove?: (accessor: string) => void;
 
   // Some dimension editors will allow users to change the operation grouping
   // from the panel, and this lets the visualization hint that it doesn't want
   // users to have that level of control
   hideGrouping?: boolean;
+
+  state: T;
 }
 
+// The only way a visualization has to restrict the query building
+export type DatasourceDimensionEditorProps<T = unknown> = DatasourceDimensionProps<T> & {
+  setState: StateSetter<T>;
+  core: Pick<CoreSetup, 'http' | 'notifications' | 'uiSettings'>;
```

Comment:
I am not sure, if we should really `Pick` properties form core here to pass down. This feels a bit arbitrary knowledge we put into the type of what we specifically need. I think if we want to pass that down, we should just pass the `core` as it is down, or use another way to access the `CoreSetup` directly from within the functions/components, e.g. via the getter/setter utils in `src/plugins/kibana_utils/common/create_getter_setter.ts` or the Kibana react contexts: src/plugins/kibana_react/README.md.

Reply:
This is not the only place we are doing this, we are mocking these specific services for testing. I don't intend to change this type in this PR.